packages/wonder-blocks-button/components/button-core.js

```diff
@@ -0,0 +1,271 @@
+                        sharedStyles.shared,
+                        sharedStyles.disabled,
+                        size === "small" && sharedStyles.small,
+                        buttonStyles.default,
+                        buttonStyles.disabled,
+                    )}
+                    disabled
+                >
+                    <LabelLarge>{this.props.children}</LabelLarge>
+                </button>
+            );
+        }
+        return (
+            <button
+                className={css(
+                    sharedStyles.shared,
+                    size === "small" && sharedStyles.small,
+                    buttonStyles.default,
+                    pressed && buttonStyles.active,
+                    (hovered || focused) && !pressed && buttonStyles.focus,
+                )}
+            >
+                <LabelLarge>{this.props.children}</LabelLarge>
+            </button>
+        );
+    }
+}
+
+const sharedStyles = StyleSheet.create({
+    shared: {
+        position: "relative",
+        height: 40,
+        margin: 0,
+        paddingLeft: 16,
+        paddingRight: 16,
+        border: "none",
+        borderRadius: 4,
+        cursor: "pointer",
+        outline: "none",
+    },
+    disabled: {
+        cursor: "auto",
+    },
+    focusRing: {
+        position: "absolute",
+        boxSizing: "border-box",
+        top: -3,
+        left: -3,
+        width: "calc(100% + 6px)",
+        height: "calc(100% + 6px)",
+        borderStyle: "solid",
+        borderWidth: 2,
+        borderColor: Color.blue,
+        borderRadius: 7,
+    },
+    focusRingRed: {
+        borderColor: Color.red,
+    },
+    focusRingLight: {
+        borderColor: Color.white,
+    },
+    small: {
+        height: 32,
+    },
+});
+
+const styles = {};
+
+const _generateStyles = (color, kind, light) => {
+    const buttonType = color + kind + light.toString();
+    if (styles[buttonType]) {
+        return styles[buttonType];
+    }
+
+    let newStyles = {};
+
+    if (kind === "primary" && !light) {
+        newStyles = {
+            default: {
+                background: color,
+                color: Color.white,
+            },
+            focus: {
+                ":before": {
+                    content: "''",
+                    position: "absolute",
+                    top: -3,
+                    left: -3,
+                    right: -3,
+                    bottom: -3,
+                    borderColor: color,
+                    borderWidth: 2,
+                    borderStyle: "solid",
+                    borderRadius: 7,
+                },
+            },
+            active: {
+                background: mix(Color.offBlack32, color),
+                color: mix(fade(color, 0.32), Color.white),
+            },
+            disabled: {
+                background: Color.offBlack32,
+                color: Color.white64,
+            },
+        };
+    } else if (kind === "primary" && light) {
```

Comment:
I think this could be collapsed with the `if` block above by using ternaries for the properties that are different, e.g.
```
newStyles = {
   default: {
      backgroundColor: light ? Color.white : color,
      color: light ? color : Color.white,
   },
   ...
```

Reply:
Should probably keep `primary`, `secondary`, `tertiary` separate though.